frontend/src/lib/utils.tsx

```diff
@@ -384,7 +384,7 @@ export function slugify(text: string): string {
         .replace(/--+/g, '-')
 }
 
-export function humanFriendlyDuration(d: string | number): string {
+export function humanFriendlyDuration(d: string | number, maxUnits?: number): string {
```

Comment:
So dayjs has [this](https://day.js.org/docs/en/durations/humanize) but it spells out the full units and also returns e.g. "a minute" instead of "1 minute". We could use dayjs token formatting instead, but then we'd likely end up with a lot of `1d 0h 0m 0s` junk to filter out with string methods.

Anyway, documented what this does and rectified the trailing space thing.